Content.Server/Destructible/DestructibleSystem.BenchmarkHelpers.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Content.Shared.Damage;
+
+namespace Content.Server.Destructible;
+
+public sealed partial class DestructibleSystem
+{
+    /// <summary>
+    /// Tests all triggers in a DestructibleComponent to see how expensive it is to query them.
+    /// </summary>
+    public void TestAllTriggers(List<Entity<DamageableComponent, DestructibleComponent>> destructibles)
+    {
+        foreach (var (uid, damageable, destructible) in destructibles)
+        {
+            foreach (var threshold in destructible.Thresholds)
+            {
+                // Chances are, none of these triggers will pass! But we have the extra code just in case!
+                if (Triggered(threshold, (uid, damageable)))
+                {
+                    RaiseLocalEvent(uid, new DamageThresholdReached(destructible, threshold), true);
+                    Execute(threshold, uid);
+                }
```

Comment:
```suggestion
                Reached(threshold, (uid, damageable)))
```
This is still the same as before? Like I mentioned, running Execute here does not make any sense because none of the thresholds will be reached without taking damage before.
So it's much better to separate that out and only test the part we are interested of. The combined situation is already tested in the test case where you deal damage to them.